app/services/samples/metadata/update_service.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+module Samples
+  module Metadata
+    # Service used to Update Samples::Metadata
+    class UpdateService < BaseService
+      SampleMetadataUpdateError = Class.new(StandardError)
+      attr_accessor :sample, :metadata, :analysis_id
+
+      def initialize(project, sample, user = nil, params = {})
+        super(user, params)
+        @project = project
+        @sample = sample
+        @metadata = params['metadata']
+        @analysis_id = params['analysis_id']
+      end
+
+      def execute # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Metrics/MethodLength
+        authorize! sample.project, to: :update_sample?
+        metadata_fields_update_status = { updated: [], not_updated: [] }
+
+        if @project.id != @sample.project.id
```

Comment:
**nitpick (blocking):** Can you move this if statement into a private method named `validate_sample_in_project`.

Reply:
Fixed in b78174939f9a80456d3eb8fbc8342885608f718e